### PR TITLE
PATH: Feature/holding tags panel support

### PR DESCRIPTION
**Pull request by mlampert**
makes holding tags "tolerant" to Arch.PanelSheet - however it does not change the overall approach and the fragility that comes with it. Although there are some minor stability improvements.
Tested changes with OCC7, which seems to be less precise and produces edges of a face that are harder to connect (needed to lower the precision - or increase the tolerance, depending on how you want to look at it).
Changed the "Add" for holding tags to be sticky - so the user can continue to add new tags, until they either hit the "Done" button or press Esc. In order to process this correctly the modifyStandardButtons is now properly routed to python task dialogs 

Adds template support for jobs which are stored in an xml file. In order to organize the user's marcros and configurations better a path specific directory is introduced (see preferences). All external files (templates, post and pre processors) are searched in the Path specific path, the macro directory and finally Path's source directory.
Existing jobs can be exported as templates from their context menu. The current approach sticks with the file name convention which is job_.....xml

Fixed PathPostProcessor accordingly.

Renamed PathLoadTool to PathToolController. It also does no longer incorporate a tooltable but stores a Tool directly. Needless to say that Path::PropertyTool is now a valid type.

Also fixed the Escape handling of PathJob and PathDressupHoldingTags if the input focus is not on the TaskPanel.

And finally, I got tired of navigating through the menus so I added all dressups to the context menu of profile and contour Paths.

### CODE REVIEW

**Comment by wwmayer** [comment]
https://forum.freecadweb.org/viewtopic.php?f=15&t=23027

File: src/Mod/Path/App/TooltablePyImp.cpp [hunk 1]
```diff
@@ -328,6 +328,22 @@ PyObject* ToolPy::copy(PyObject * args)
 }
 
 
+PyObject* ToolPy::fromTemplate(PyObject * args)
+{
+    PyObject *pcObj;
+    if (PyArg_ParseTuple(args, "S", &pcObj)) {
```
Comment (wwmayer) [inline src/Mod/Path/App/TooltablePyImp.cpp]: Is there a reason not to use 's' (lower-case) as we do it everywhere else? Now that FreeCAD is mostly ported to Py3 using lower-case "s" won't cause extra porting effort while upper-case "S" will do:
https://docs.python.org/3/c-api/arg.html (search for bytes object)

Btw, the extra check with PyString_Check is superfluous as the docs say that if it's not a str a TypeError will be raised:
https://docs.python.org/2/c-api/arg.html

For performance reason you should not raise an exception here but instead use PyErr_SetString(PyExc_TypeError, ...) and return 0

**Comment by wwmayer** [comment]
in the forum thread you wrote:

> due to an internal refactoring (renaming LoadTool to PathToolController) all saved jobs will throw exceptions when loaded - you will have to recreate them. - So much for the bad news.

So, is this a major or only a minor thing? How does the other Path devs think about it? Is it acceptable?

File: src/Mod/Path/App/TooltablePyImp.cpp (hunk 1 above)
Comment (mlampert) [inline src/Mod/Path/App/TooltablePyImp.cpp]: Great - thanks for the info, will change and update the PR.

**Comment by mlampert** [comment]
Sliptonic and I talked about the refactoring and he wanted to rename the class for quite a while as well.

**Comment by mlampert** [comment]
Also - I commented out the printf statements in Observer since they were obscuring everything else during debugging - forgot to revert that. Do you want me to put them back in place or did they serve their purpose and can stay commented out?

**Comment by wwmayer** [comment]
You can leave them commented out.

**Pull request merged**